django/core/validators.py

```diff
@@ -113,11 +115,24 @@ def __init__(self, schemes=None, **kwargs):
 
     def __call__(self, value):
         value = force_text(value)
-        # Check first if the scheme is valid
         scheme = value.split('://')[0].lower()
-        if scheme not in self.schemes:
+
+        # A naive check for valid schemes
+        # schemes may contain an optional +, but only if followed
+        # by one or more characters
+        # RFC 3986 has the complete details
+        # http://tools.ietf.org/html/rfc3986#section-3.1
+        scheme_regex = _lazy_re_compile(r'^[a-z][a-z0-9]+(\+[a-z0-9]+)?$')
```

Comment:
Actually the idea is to do this validation only if `schemes == '__all__'`. If the user provides some other scheme, then we might as well allow it.